synphot/models.py

```diff
@@ -495,12 +506,21 @@ def __init__(self, *args, **kwargs):
         gaussian_amp_to_totflux = np.sqrt(2.0 * np.pi) * self.stddev
 
         if total_flux is None:
+            u_str = 'PHOTLAM'
             total_flux = self.amplitude * gaussian_amp_to_totflux
         else:
-            self.amplitude = total_flux / gaussian_amp_to_totflux
+            u_str = 'FLAM'
+            # total_flux is passed in unaltered, any conversion error would
+            # happen here.
+            tf_unit = u.erg / (u.cm * u.cm * u.s)
+            if isinstance(total_flux, u.Quantity):
+                total_flux = total_flux.to(tf_unit)
```

Comment:
Could this conversion to `tf_unit` include an equivalency parameter so that total fluxes can be provided in units of, e.g., photons/s/cm2 or Jy Hz?

Reply:
@mkelley , I thought about that but I wasn't sure if using `u.spectral_density(mean)` would give the correct conversion. If you think that is correct, then I can certainly add it in.

Reply:
No, wait. I take it back. I can't use `spectral_density` because it is integrated flux unit.

Reply:
Ah, yes.  I was thinking there was an equivalency for flux.

Reply:
How about we merge this as is and see how things play out over at astropy/astropy#7593?

Reply:
Great!